polkadot/node/core/dispute-coordinator/src/lib.rs

```diff
@@ -286,30 +288,43 @@ impl DisputeCoordinatorSubsystem {
 		ChainScraper,
 		SessionIndex,
 		bool,
+		initialized::OffchainDisabledValidators,
 	)> {
 		let now = clock.now();
 
-		let active_disputes = match overlay_db.load_recent_disputes() {
-			Ok(disputes) => disputes
-				.map(|disputes| get_active_with_status(disputes.into_iter(), now))
-				.into_iter()
-				.flatten(),
+		// We assume the highest session is the passed leaf. If we can't get the session index
+		// we can't initialize the subsystem so we'll wait for a new leaf
+		let highest_session = runtime_info
+			.get_session_index_for_child(ctx.sender(), initial_head.hash)
+			.await?;
+		let earliest_session = highest_session.saturating_sub(DISPUTE_WINDOW.get() - 1);
+
+		// Load recent disputes from the database
+		let recent_disputes = match overlay_db.load_recent_disputes() {
+			Ok(disputes) => disputes.unwrap_or_default(),
 			Err(e) => {
 				gum::error!(target: LOG_TARGET, "Failed initial load of recent disputes: {:?}", e);
 				return Err(e.into())
 			},
 		};
 
-		// We assume the highest session is the passed leaf. If we can't get the session index
-		// we can't initialize the subsystem so we'll wait for a new leaf
-		let highest_session = runtime_info
-			.get_session_index_for_child(ctx.sender(), initial_head.hash)
-			.await?;
+		// Initialize offchain disabled validators from recent disputes
+		let offchain_disabled_validators = initialized::OffchainDisabledValidators::new_from_state(
```

Comment:
Nit: debug/trace with number of disabled validators  at startup. 

Do we have already have some log that tells us which validators are off-chain disabled?

Reply:
Yeah, i think we should even do info logs for public shaming :P